mongo/cursor.go

```diff
@@ -185,11 +185,17 @@ func (c *Cursor) Close(ctx context.Context) error {
 // This method requires driver version >= 1.1.0.
 func (c *Cursor) All(ctx context.Context, results interface{}) error {
 	resultsVal := reflect.ValueOf(results)
+
 	if resultsVal.Kind() != reflect.Ptr {
 		return errors.New("results argument must be a pointer to a slice")
 	}
 
 	sliceVal := resultsVal.Elem()
+
```

Comment:
nit: Remove this blank line. We try to avoid single-line blocks and the conditional below fits into the current block.

Reply:
> This looks good! It wasn't part of the original ticket, but the mgo code also has this check:
> 
> ```
> if slicev.Kind() != reflect.Slice {
> 		panic("result argument must be a slice address")
> }
> ```
> 
> right after the `sliceVal.Kind() == reflect.Interface` check you have. I think this makes sense to add. We first check if it's a pointer, then get the thing it's pointing to and potentially unpack it if it's kind is `reflect.Interface`, and finally check if the thing we ended up with is a slice. Thoughts on adding such a check in this PR?

This makes sense to me, in this function, we first check if it's a pointer, and if not we return an error. If the thing is not a slice, should we panic or return an error instead?